manifests/agent.pp

```diff
@@ -20,11 +28,85 @@
     },
   },
 ) {
-  require hyperglass::hyperglassdir
-  contain hyperglass::agent::install
-  contain hyperglass::agent::config
-  contain hyperglass::agent::service
-  Class['hyperglass::agent::install']
-  -> Class['hyperglass::agent::config']
-  ~> Class['hyperglass::agent::service']
+  file { '/opt/hyperglass':
```

Comment:
what if someone doesn't install the agent on one particular machine?

then this doesn't get created, and then the other directory creations fail too

Reply:
The server includes the agent so it will always be there.

Reply:
Actually refactored as per Tim and now this common code is in the hyperglass class.